src/governance.ts

```diff
@@ -128,10 +133,51 @@ export class GovernanceCanister {
     return this.responseConverters.toListProposalsResponse(rawResponse);
   };
 
+  public stakeNeuron = async ({
+    stake,
+    principal,
+    ledgerCanister,
+  }: {
+    stake: ICP;
+    principal: Principal;
+    ledgerCanister: LedgerCanister;
+  }): Promise<void> => {
+    if (stake.toE8s() < E8S_PER_ICP) {
+      throw "Need a minimum of 1 ICP to stake a neuron";
+    }
+
+    try {
+      const nonceBytes = new Uint8Array(randomBytes(8));
+      const nonce = uint8ArrayToBigInt(nonceBytes);
+      const toSubAccount = buildSubAccount(nonceBytes, principal);
+      const accountIdentifier = AccountIdentifier.fromPrincipal({
+        principal: this.canisterId,
+        subAccount: toSubAccount,
+      });
+
+      // Send amount to the ledger.
+      await ledgerCanister.transfer({
+        memo: nonce,
+        amount: stake,
+        to: accountIdentifier,
+        // TODO: Support subaccounts
+      });
+
+      // Notify the governance of the transaction so that the neuron is created.
+      await this.claimOrRefreshNeuronFromAccount({
+        controller: principal,
+        memo: nonce,
+      });
+    } catch (err) {
+      // TODO: Manage errors
+      console.error(err);
+    }
+  };
+
   /**
    * Gets the NeuronID of a newly created neuron.
    */
-  public claimOrRefreshNeuronFromAccount = async (
+  private claimOrRefreshNeuronFromAccount = async (
```

Comment:
I don't think there is harm in keeping this method public. This class should mimic the governance canister, and so anything that is public in the governance canister should be public here as well.

Reply:
ok